### PR TITLE
fit the latest version of predis

### PR DESCRIPTION
**Pull request by geniuslinchao**
The latest version of 'predis' use 'getCommandFactory' function to replace 'getProfile'

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,18 +1,18 @@
 {
-    "name": "jamescauwelier/psredis",
+    "name": "geniuslinchao/psredis",
```
Comment (jamescauwelier) [inline composer.json]: Please keep the original namespace alive?

File: composer.json [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+            "name": "James Cauwelier(origin)/geniuslinchao",
+            "email": "james.cauwelier@gmail.com(origin)/geniuslinchao@gmail.com",
             "role": "Developer"
         }
     ],
     "license": "MIT",
     "require": {
-        "php": ">=5.4",
+        "php": ">=5.5",
         "predis/predis": ">=1.0"
```
Comment (jamescauwelier) [inline composer.json]: Can you edit these requirements so that no incompatible predis versions can be used?

**Comment by jamescauwelier** [comment]
@geniuslinchao thanks you so much for your work on this.  Let's try to get it merged as soon as possible!  I have left a few comments in the code for you to fix before we can do the merge.

Thank you!

**Comment by nrk** [comment]
@geniuslinchao just to be clear that's from the **development** version of Predis, kind of different from being the current latest. The latest **stable** release of Predis as the time of writing is `v1.1.0` and being a minor release the API is compatible with previous releases in the `v1.0` branch. The changes you are speaking about are related to the development of the next major release which still has a long way to go and its API is a moving target as it's far from being definitive (I just started working on it a couple of weeks ago).

File: composer.json (hunk 1 above)
Comment (geniuslinchao) [inline composer.json]: Sorry, it's a mistake:)

**Comment by geniuslinchao** [comment]
@nrk You mean we don't  need to modify the code, but just change required version of predis to ">=1.0 <2.0", right?

File: composer.json (hunk 2 above)
Comment (geniuslinchao) [inline composer.json]: Yes, this is a better choice.

**Comment by nrk** [comment]
@geniuslinchao yes it's simply a matter of changing the requirement to `~1.0` (which acts exactly like `>=1.0 <2.0.0` in this case, [see the docs](https://getcomposer.org/doc/articles/versions.md#tilde)). I think you should close this PR and open a new one or rebase all of the commits into one with only this change to keep the noise down (too many back-and-forth commits in the history at this point), but that's up to you and @jamescauwelier.

**Comment by jamescauwelier** [comment]
I love simple solutions! 

@geniuslinchao Let's do a new PR as suggested by @nrk ?

**Comment by geniuslinchao** [comment]
ok! @jamescauwelier

**Pull request closed without merging**